src/main/datalog_console/components/query.cljs

```diff
@@ -29,43 +29,49 @@
                                                    2 (reverse (sort result)))))]]])))
 
 (defn query []
-  (let [saved-query (localstorage/get-item (str ::query-text))
-        query-text (r/atom (or saved-query (get example-queries "All attributes")))
+  (let [saved-query (r/atom ::loading)
+        _ (chromestorage/get-item ::query-text (fn [result] (reset! saved-query (goog.object/get result (str ::query-text)))))
```

Comment:
We should probably move this after the track on line 37. This way everything is wired up before we call get-item making it a bit more readable and avoiding potential race conditions

Reply:
Yea this makes sense.